### PR TITLE
chore: add fmt for statement query

### PR DESCRIPTION
**Pull request by paomian**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

* impl `Display` trait for Statement Query

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Review by waynexia — APPROVED** [review]

**Comment by killme2008** [comment]
Missing unit tests.

File: src/datatypes/src/data_type.rs [hunk 1]
```diff
@@ -62,6 +63,32 @@ pub enum ConcreteDataType {
...
+            ConcreteDataType::UInt64(_) => write!(f, "UInt64"),
+            ConcreteDataType::Float32(_) => write!(f, "Float32"),
+            ConcreteDataType::Float64(_) => write!(f, "Float64"),
+            ConcreteDataType::Binary(_) => write!(f, "Binary"),
+            ConcreteDataType::String(_) => write!(f, "String"),
+            ConcreteDataType::Date(_) => write!(f, "Date"),
+            ConcreteDataType::DateTime(_) => write!(f, "DateTime"),
+            ConcreteDataType::Timestamp(_) => write!(f, "Timestamp"),
+            ConcreteDataType::List(_) => write!(f, "List"),
+            ConcreteDataType::Dictionary(_) => write!(f, "Dictionary"),
```
[the remark this answers is not in the document]
Comment (paomian) [reply src/datatypes/src/data_type.rs:87]: I think It is not needed now.

**Review by killme2008 — APPROVED** [review]
LGTM

**Comment by sunng87** [comment]
Perhaps we can use `self.name()` for `fmt` implementation so that there is no need to write these names twice

**Pull request merged**